lib/Geo/GeoJson.cpp

```diff
@@ -39,616 +41,750 @@
 #include "Basics/VelocyPackHelper.h"
 #include "Basics/debugging.h"
 #include "Geo/GeoParams.h"
+#include "Geo/ShapeContainer.h"
 #include "Geo/S2/S2MultiPointRegion.h"
 #include "Geo/S2/S2MultiPolyline.h"
-#include "Geo/ShapeContainer.h"
 #include "Logger/Logger.h"
 
+namespace arangodb::geo::json {
 namespace {
-// Have one of these values:
-std::string const kTypeStringPoint = "Point";
-std::string const kTypeStringLineString = "LineString";
-std::string const kTypeStringPolygon = "Polygon";
-std::string const kTypeStringMultiPoint = "MultiPoint";
-std::string const kTypeStringMultiLineString = "MultiLineString";
-std::string const kTypeStringMultiPolygon = "MultiPolygon";
-std::string const kTypeStringGeometryCollection = "GeometryCollection";
-}  // namespace
 
-namespace {
-inline bool sameCharIgnoringCase(char a, char b) noexcept {
-  return arangodb::basics::StringUtils::toupper(a) ==
-         arangodb::basics::StringUtils::toupper(b);
+// The main idea is check in our CI 3 compile time branches:
+// 1. Validation true
+// 2. Validation false kIsMaintainer true
+// 3. Validation false kIsMaintainer false
+#if defined(ARANGODB_ENABLE_MAINTAINER_MODE) && defined(__linux__) && \
+    !defined(__aarch64__)
```

Comment:
It's not no validation in maintainer mode on arm, it's needed to test other code path when user specify no validation

Reply:
I do not understand this argument. For my money, this `!defined(__aarch64__)` means that if I compile in maintainer mode on Linux ARM, then I will get
```
constexpr bool kIsMaintainer = false;
```
Therefore, validation will not be turned on in maintainer mode on ARM in cases where it is usually (for efficiency reasons) turned off like this one:
```
template<bool Validation>
Result parseRegion(velocypack::Slice vpack, ShapeContainer& region,
                   std::vector<S2Point>& cache, bool legacy) {
  auto r = parseRegionImpl<(Validation || kIsMaintainer)>(vpack, region, cache,
                                                          legacy);
  TRI_ASSERT(Validation || r.ok()) << r.errorMessage();
  return r;
}
```
Why this difference on ARM? If you want to expose this (unvalidated) code path in tests and use the "ARM" builds for this, then please add a comment to explain this, because otherwise a reader will think "Is there anything which is not working on ARM and is hidden by this condition?".

Reply:
Ok, first of all this about Validation and isMaintainer.

User of this function now only about Validation, and then he used parseRegion/etc. He pass Validation parameter.

If passed Validation == false shouldn't be any error, but at the same time shouldn't be any checks (because performance reasons).

So I decide to make Validation == false in Maintainer Mode same as Validation == true + assert which assume no errors was.

But our code only compiled  (and provided to user) but not tested without maintainer. So I decide to don't have enable my checks on arm, to see other code path also works


Excuse me, I though it was clear from comment

Reply:
OK, now I understand. Then let's add a comment in the code so that others understand, too.